gremlin-go/example/example.go

```diff
@@ -21,38 +21,137 @@ package main
 
 import (
 	"fmt"
+
 	gremlingo "github.com/apache/tinkerpop/gremlin-go/driver"
-	"github.com/apache/tinkerpop/gremlin-go/v3/driver"
 )
 
 // syntactic sugar
 var __ = gremlingo.T__
 var gt = gremlingo.P.Gt
 var order = gremlingo.Order
+var T = gremlingo.T
+var P = gremlingo.P
 
 func main() {
-	// Creating the connection to the server.
-	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
+	connectionExample()
+	basicGremlinExample()
+	modernTraversalExample()
+}
+
+func connectionExample() {
+	// Creating the connection to the server
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+
+	// Connecting to the server with customized configurations
+	driverRemoteConnection, err = gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
 		func(settings *gremlingo.DriverRemoteConnectionSettings) {
-			settings.TraversalSource = "gmodern"
+			settings.TraversalSource = "g"
+			settings.NewConnectionThreshold = 4
+			settings.EnableCompression = false
+			settings.ReadBufferSize = 0
+			settings.WriteBufferSize = 0
 		})
+
+	// Error handling
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
+
 	// Cleanup
 	defer driverRemoteConnection.Close()
 
-	// Creating graph traversal
+	// Creating the graph traversal
+	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
+	fmt.Println(g)
+}
+
+func basicGremlinExample() {
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer driverRemoteConnection.Close()
 	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
 
-	// Perform traversal
-	result, err := g.V().HasLabel("person").Has("age", __.Is(gt(28))).Order().By("age", order.Desc).Values("name").ToList()
+	// Basic Gremlin: adding and retrieving data
+	v1, err := g.AddV("person").Property("name", "marko").Next()
```

Comment:
In general, best practice in go would be to actually check `err` in an if statement after each of these and handle it somehow (print it, return it...). I think it's fine to skip that error handling here to keep the example concise.

FYI, normally if you were to skip the error handling, you would want to write this like
```v1, _ := g.AddV("person").Property("name", "marko").Next()```
to be explicit that err is unused. Once again I think it's fine as is here so that it is clear in the example that the second returned value is the error.